src/components/views/admin/BuilderView.tsx

```diff
@@ -1,156 +1,243 @@
+      dataObject: dataObj,
+      targetType: type,
+    });
+
+    Object.keys(typeDataByMemberId).forEach((memberId) => {
+      if (memberId in candidatesCv) {
+        candidatesCv[memberId][type] = typeDataByMemberId[memberId].map(
+          (item) => item.data,
+        );
+      } else {
+        candidatesCv[memberId] = {
+          [type]: typeDataByMemberId[memberId].map((item) => item.data),
+        };
+      }
+    });
+  });
+
+  const [ratings, setRatings] = useState<{ [key: string]: number | null }>({});
+  const handleRatingChange = (
+    memberId: string,
+    newRating: number | null,
+  ): void => {
+    const rateData = appDataArray.find(
+      (data) =>
+        data.data.memberId === memberId && data.type === 'candidateRateInfo',
+    );
+    if (rateData) {
+      handlePatch(rateData.id, {
+        ...rateData.data,
+        rating: newRating,
+        memberId,
+      });
+    } else {
+      handlePost({ memberId, rating: newRating });
+    }
+    setRatings((prevRatings) => ({
+      ...prevRatings,
+      [memberId]: newRating,
+    }));
+  };
+
+  const handleDownload = async (memberId: string): Promise<void> => {
+    const candidateCvData = candidatesCv[memberId];
+    const { component: CvTemplate } = TEMPLATES.find(
+      (t) => t.id === candidateCvData.cvStatusInfo.selectedTemplateId,
+    ) || { component: FirstTemplate };
+    const renderedTemplate = <CvTemplate cvValues={candidateCvData} />;
+    const pdfBlob = await pdf(renderedTemplate).toBlob();
+    if (!candidateCvData.cvStatusInfo.customCv) {
+      saveAs(pdfBlob, 'generated-cv.pdf');
+    }
+  };
+  const [visibility, setVisibility] = useState<string | null>(null);
+  const [renderedFile, setRenderedFile] = useState<string | null>(null);
+  const handleView = async (memberId: string): Promise<void> => {
+    if (visibility === memberId) {
+      setVisibility(null);
+      setRenderedFile(null);
+    } else {
+      const candidateCvData = candidatesCv[memberId];
+      const { component: CvTemplate } = TEMPLATES.find(
+        (t) => t.id === candidateCvData.cvStatusInfo.selectedTemplateId,
+      ) || { component: FirstTemplate };
+      const fileToRender = <CvTemplate cvValues={candidateCvData} />;
+      const pdfBlob = await pdf(fileToRender).toBlob();
+      const dataUrl = URL.createObjectURL(pdfBlob);
+      setVisibility(memberId);
+      setRenderedFile(dataUrl);
+      window.open(dataUrl, '_blank');
+    }
+  };
+  const handleDownloadAll = async (): Promise<void> => {
+    const zip = new JSZip();
+    const candidateIds = Object.keys(candidatesCv);
+
+    // Use Promise.all with map to wait for all async operations to complete
+    await Promise.all(
+      candidateIds.map(async (memberId) => {
+        const candidateCvData = candidatesCv[memberId];
+        const { component: CvTemplate } = TEMPLATES.find(
+          (t) => t.id === candidateCvData.cvStatusInfo.selectedTemplateId,
+        ) || { component: FirstTemplate };
```

Comment:
Here you assign the FirstTemplate to users that may have uploaded their own cv ?

Maybe we should skip users that haven't selected a cv template, and simply supply the Cv they have uploaded ?

Reply:
This part still needs to be more handled, since the candidates might upload their own CV, if we wanted to skil that part as you mentioned and simply supply the CV they have uploaded, I think there might be an issue of handling those who just selected a template, since the Reviewer might also want to see their generated CV PDF file.